core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -575,8 +586,9 @@ private void maybeTerminateChannel() {
       return;
     }
     if (shutdown.get() && subchannels.isEmpty() && oobChannels.isEmpty()) {
-      log.log(Level.FINE, "[{0}] Terminated", getLogId());
+      logger.log(Level.FINE, "[{0}] Terminated", getLogId());
       terminated = true;
+      phantom.terminated = true;
```

Comment:
It'd probably be good to `phantom.clear()` and/or `refs.remove(phatom)` in this case.

Reply:
Done.